### PR TITLE
fix: certificate_id should be optional for api_gateway resource

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: services/apigateway/apigateway.go [hunk 1]
```diff
@@ -155,13 +155,13 @@ func setGatewayConfig(d *schema.ResourceData) apigateway.Gateway {
...
 		domainData := domain.(map[string]interface{})
 		name := domainData["name"].(string)
-		certificateID := domainData["certificate_id"].(string)
 
 		customDomainObj := apigateway.GatewayCustomDomains{
-			Name:          &name,
-			CertificateId: &certificateID,
+			Name: &name,
+		}
+		if cert, ok := domainData["certificate_id"]; ok && cert != "" {
```
Comment (cbeti-ionos) [inline services/apigateway/apigateway.go:162]: not sure if `cert != ""` is still needed when the field has `validation.IsUUID`

**Review by cbeti-ionos — APPROVED** [review]

File: ionoscloud/resource_apigateway_test.go [hunk 2]
```diff
@@ -95,8 +83,8 @@ func TestAccAPIGateway_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("ionoscloud_apigateway.example", "name", "example"),
 					resource.TestCheckResourceAttr("ionoscloud_apigateway.example", "logs", "false"),
 					resource.TestCheckResourceAttr("ionoscloud_apigateway.example", "metrics", "true"),
-					resource.TestCheckResourceAttr("ionoscloud_apigateway.example", "custom_domains.0.name", "example.com"),
-					resource.TestCheckResourceAttr("ionoscloud_apigateway.example", "custom_domains.0.certificate_id", "00000000-0000-0000-0000-000000000000"),
+					// can't be used in tests, as it requires a reachable domain name
+					// resource.TestCheckResourceAttr("ionoscloud_apigateway.example", "custom_domains.0.name", "example.com"),
```
Comment (digna-ionos) [inline ionoscloud/resource_apigateway_test.go:87]: why we don't remove that line instead of commenting?

**Review by avirtopeanu-ionos — DISMISSED** [review]

File: services/apigateway/apigateway.go (hunk 1 above)
Comment (cristiGuranIonos) [reply services/apigateway/apigateway.go:162]: it is, for some reason it's set to "" if you don't set anything

File: ionoscloud/resource_apigateway_test.go (hunk 2 above)
Comment (cristiGuranIonos) [reply ionoscloud/resource_apigateway_test.go:87]: we might want to add it back if we create an externally available name. we can remove it

**Pull request merged**